### PR TITLE
fix(components): [table] fixed column supported in grouped header

### LINKED ISSUE #3670: [bug report] eltable fixed multiple columns and header grouping have bugs when they are used at the same time
### Element Plus version
1.1.0-beta.16

### OS/Browsers version
macOS / Google Chrome 94.0.4606.61（正式版本） (x86_64)

### Vue version
3.2.19

### Reproduction Link
https://codesandbox.io/s/element-plus-spa-forked-03nfn

### Steps to reproduce
配置好 element-plus环境，使用 ElTable组件后，同时使用表头分组和列固定

### What is Expected?
同时使用表头分组和列固定，所有列正常显示

### What is actually happening?
部分列将无法显示

### LINKED ISSUE #6949: [Bug Report] [Component] [table, table-column] 多级表头el-table-column嵌套el-table-column时使用fixed会失效
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.1.7`
- Browser / OS: `UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.82 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-table`
- `el-table-column`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGUgOmRhdGE9XCJ0YWJsZURhdGFcIiBzdHlsZT1cIndpZHRoOiAxMDAlXCI+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiZGF0ZVwiIGxhYmVsPVwiRGF0ZVwiIHdpZHRoPVwiMTUwXCIgZml4ZWQ9XCJsZWZ0XCIgPlxuICAgICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJzdGF0ZVwiIGxhYmVsPVwiU3RhdGVcIiB3aWR0aD1cIjEyMFwiIC8+XG4gICAgPC9lbC10YWJsZS1jb2x1bW4+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIkRlbGl2ZXJ5IEluZm9cIj5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIk5hbWVcIiB3aWR0aD1cIjEyMFwiIC8+XG4gICAgICBcbiAgICA8L2VsLXRhYmxlLWNvbHVtbj5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiQWRkcmVzcyBJbmZvXCI+XG4gICAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImNpdHlcIiBsYWJlbD1cIkNpdHlcIiB3aWR0aD1cIjUyMFwiIC8+XG4gICAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImFkZHJlc3NcIiBsYWJlbD1cIkFkZHJlc3NcIiAvPlxuICAgICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJ6aXBcIiBsYWJlbD1cIlppcFwiIHdpZHRoPVwiNjIwXCIgLz5cbiAgICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICA8L2VsLXRhYmxlPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmNvbnN0IHRhYmxlRGF0YSA9IFtcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAzJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDInLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDgnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTA3JyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XCJpbXBvcnRzXCI6e319In0=)

### Steps to reproduce

<el-table :data="tableData" style="width: 100%">
    <el-table-column prop="date" label="Date" width="150" fixed="left" >
        <el-table-column prop="day" label="Day" width="520" fixed="left"/>
    </el-table-column>
    <el-table-column prop="Info" label="Delivery Info">
      <el-table-column prop="name" label="Name" width="520" />
    </el-table-column>
</el-table>

### What is Expected?

拖动横向滚动条列头和数据列同时固定

### What is actually happening?

拖动横向滚动条只有最顶层的列头固定，嵌套的列头和数据量不能固定

### Additional comments

_(empty)_

### LINKED ISSUE #9216: [Component] [table] el-table组件 多级表头fix属性 没有被固定住
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.12`
- Browser / OS: `chrome`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[CodePen](https://codepen.io/a563346904/pen/QWmrwbm)

### Steps to reproduce

```html
<html>

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width,initial-scale=1.0" />
  <script src="https://unpkg.com/vue@next"></script>
  <!-- import CSS -->
  <link rel="stylesheet" href="https://unpkg.com/element-plus/dist/index.css">
  <!-- import JavaScript -->
  <script src="https://unpkg.com/element-plus"></script>
  <title>Element Plus demo</title>
</head>

<body>
  <div id="app">
    <el-table :data="tableData" height="500px" style="width: 800px">
      <el-table-column label="a" prop="a" key="a" fixed></el-table-column>
      <el-table-column label="bbb" prop="b" key="b" width="300" fixed="left">
        <el-table-column label="123" prop="d" width="200"></el-table-column>
        <el-table-column label="qwe" prop="d" width="200"></el-table-column>
      </el-table-column>
      <el-table-column label="cccc" prop="c" key="c" min-width="400"></el-table-column>
      <el-table-column label="ddddd" prop="d" key="d" width="200" fixed="right"></el-table-column>
    </el-table>
  </div>
  <script>
    const App = {
      data() {
        return {
          message: "Hello Element Plus",
          tableData: new Array(1).fill({ a: 1, b: 2, c: 3, d: 4 })
        };
      },
    };
    const app = Vue.createApp(App);
    app.use(ElementPlus);
    app.mount("#app");
  </script>
</body>

</html>
```

### What is Expected?

多级fixed的表头固定

### What is actually happening?

多级fixed的表头没有固定

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by tinyfind**
close #9216,close #3670,close #6949 

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by tolking** [comment]
[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGUgOmRhdGE9XCJ0YWJsZURhdGFcIiBzdHlsZT1cIndpZHRoOiAxMDAlXCI+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIkFkZHJlc3MgSW5mb1wiIGZpeGVkPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwic3RhdGVcIiBsYWJlbD1cIlN0YXRlXCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiY2l0eVwiIGxhYmVsPVwiQ2l0eVwiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICA8L2VsLXRhYmxlLWNvbHVtbj5cbiAgICBcbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImFkZHJlc3NcIiBsYWJlbD1cIkFkZHJlc3NcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJ6aXBcIiBsYWJlbD1cIlppcFwiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cInN0YXRlXCIgbGFiZWw9XCJTdGF0ZVwiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImNpdHlcIiBsYWJlbD1cIkNpdHlcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJhZGRyZXNzXCIgbGFiZWw9XCJBZGRyZXNzXCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiemlwXCIgbGFiZWw9XCJaaXBcIiB3aWR0aD1cIjEwMFwiIC8+XG5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiQWRkcmVzcyBJbmZvXCI+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJzdGF0ZVwiIGxhYmVsPVwiU3RhdGVcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJjaXR5XCIgbGFiZWw9XCJDaXR5XCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiYWRkcmVzc1wiIGxhYmVsPVwiQWRkcmVzc1wiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cInppcFwiIGxhYmVsPVwiWmlwXCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJBZGRyZXNzIEluZm9cIiBmaXhlZD1cInJpZ2h0XCI+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJzdGF0ZVwiIGxhYmVsPVwiU3RhdGVcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJjaXR5XCIgbGFiZWw9XCJDaXR5XCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICA8L2VsLXRhYmxlPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmNvbnN0IHRhYmxlRGF0YSA9IFtcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAzJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDInLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDgnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTA3JyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuXVxuXG48L3NjcmlwdD5cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMDA5Ni1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vdW5wa2cuY29tL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTAwOTYtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL3VucGtnLmNvbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTEwMDk2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTEwMDk2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

[before](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGUgOmRhdGE9XCJ0YWJsZURhdGFcIiBzdHlsZT1cIndpZHRoOiAxMDAlXCI+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIkFkZHJlc3MgSW5mb1wiIGZpeGVkPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwic3RhdGVcIiBsYWJlbD1cIlN0YXRlXCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiY2l0eVwiIGxhYmVsPVwiQ2l0eVwiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICA8L2VsLXRhYmxlLWNvbHVtbj5cbiAgICBcbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImFkZHJlc3NcIiBsYWJlbD1cIkFkZHJlc3NcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJ6aXBcIiBsYWJlbD1cIlppcFwiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cInN0YXRlXCIgbGFiZWw9XCJTdGF0ZVwiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImNpdHlcIiBsYWJlbD1cIkNpdHlcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJhZGRyZXNzXCIgbGFiZWw9XCJBZGRyZXNzXCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiemlwXCIgbGFiZWw9XCJaaXBcIiB3aWR0aD1cIjEwMFwiIC8+XG5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiQWRkcmVzcyBJbmZvXCI+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJzdGF0ZVwiIGxhYmVsPVwiU3RhdGVcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJjaXR5XCIgbGFiZWw9XCJDaXR5XCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiYWRkcmVzc1wiIGxhYmVsPVwiQWRkcmVzc1wiIHdpZHRoPVwiMTAwXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cInppcFwiIGxhYmVsPVwiWmlwXCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJBZGRyZXNzIEluZm9cIiBmaXhlZD1cInJpZ2h0XCI+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJzdGF0ZVwiIGxhYmVsPVwiU3RhdGVcIiB3aWR0aD1cIjEwMFwiIC8+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJjaXR5XCIgbGFiZWw9XCJDaXR5XCIgd2lkdGg9XCIxMDBcIiAvPlxuICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICA8L2VsLXRhYmxlPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmNvbnN0IHRhYmxlRGF0YSA9IFtcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAzJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDInLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDgnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTA3JyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuXVxuXG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

File: packages/components/table/src/util.ts [hunk 1]
```diff
@@ -379,6 +379,18 @@ export function createTablePopper(
   return popperInstance
 }
 
+function getCurrentColumns<T>(column: TableColumnCtx<T>): TableColumnCtx<T>[] {
+  if (column.children) {
+    return column.children.flatMap(getCurrentColumns)
```
Comment (holazz) [inline packages/components/table/src/util.ts:384]: **Suggestion**

```ts
import { flatMap } from 'lodash-unified'
```
```suggestion
    return flatMap(column.children, getCurrentColumns)
```

**Comment by tolking** [comment]
Can you add a test, and add an example for documentation?

**Comment by tinyfind** [comment]
@tolking  ok

**Comment by tinyfind** [comment]
> Can you add a test, and add an example for documentation?

ok, pleasereview again

**Comment by tolking** [comment]
Some files have conflicts that need to be resolved

File: packages/components/table/__tests__/table-column.test.ts [hunk 2]
```diff
@@ -701,6 +701,71 @@ describe('table column', () => {
...
+        },
+        template: `
+          <el-table :data="testData">
+            <el-table-column prop="name" />
+            <el-table-column label="group" fixed="left">
+              <el-table-column label="group's group">
+                <el-table-column prop="runtime" width="100" fixed="right"/>
+                <el-table-column prop="director" width="100" fixed="right"/>
+              </el-table-column>
+              <el-table-column  prop="director"/>
```
Comment (tolking) [inline packages/components/table/__tests__/table-column.test.ts:718]: There is an extra space here

Comment (tinyfind) [reply packages/components/table/__tests__/table-column.test.ts]: resolved

File: packages/components/table/__tests__/table-column.test.ts [hunk 3]
```diff
@@ -701,6 +701,71 @@ describe('table column', () => {
...
+              <el-table-column prop="runtime" width="100" fixed="left"/>
+              <el-table-column prop="director" width="50"/>
+            </el-table-column>
+            <el-table-column prop="runtime"/>
+          </el-table>
+        `,
+
+        created() {
+          this.testData = getTestData()
+          this.groupFixed = 'left'
```
[the remark this answers is not in the document]
Comment (tinyfind) [reply packages/components/table/__tests__/table-column.test.ts]: resolved

File: packages/components/table/src/util.ts [hunk 4]
```diff
@@ -387,21 +399,18 @@ export const isFixedColumn = <T>(
 ) => {
   let start = 0
   let after = index
+  const columns = store.states.columns.value
```
Comment (tolking) [inline packages/components/table/src/util.ts:402]: Is it possible to lose reactivity?

Comment (tinyfind) [reply packages/components/table/src/util.ts:402]: yes, it has no affect

**Review by tolking — APPROVED** [review]
LGTM.

**Review by holazz — APPROVED** [review]

**Pull request merged**